src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -109,27 +109,27 @@ object LoadVCF {
         (line.getType == VCFHeaderLineType.Flag && line.getCount == 0)))
       Field(id, baseType, i, attrs)
     else
-      Field(id, TArray(baseType), i, attrs)
+      Field(id, TArray(if (arrayElementsRequired) !baseType else baseType), i, attrs)
```

Comment:
I wonder if something like `baseType.setRequired(arrayElementsRequired)` isn't cleaner.  This idiom shows up quite a bit and it might be nice to have a way to write it without duplicating the expression for `baseType`.